src/Microsoft.OpenApi.Readers/OpenApiYamlDocumentReader.cs

```diff
@@ -114,11 +114,15 @@ public async Task<ReadResult> ReadAsync(YamlDocument input)
             // Validate the document
             if (_settings.RuleSet != null && _settings.RuleSet.Rules.Count > 0)
             {
-                var errors = document.Validate(_settings.RuleSet);
-                foreach (var item in errors)
+                var openApiErrors = document.Validate(_settings.RuleSet);
+                foreach (var item in openApiErrors.Where(e => e is OpenApiValidatorError))
                 {
                     diagnostic.Errors.Add(item);
```

Comment:
assuming diagnostics is a list, you could then call AddRange to avoid multiple growths and enable vectorialization

Reply:
It's an `IList` which doesn't have the `AddRange` extension method.